driver-mapping/src/test/java/com/datastax/driver/mapping/MapperAccessorParamsTest.java

```diff
@@ -85,6 +131,43 @@ void should_include_enumtype_in_accessor_string() {
         ResultSet addUserStr(int key, @Enumerated(EnumType.STRING) Enum value);
     }
 
+    /** Tests various ways to match method parameters to query bind markers. */
+    @Accessor
+    public interface UserPhoneAccessor {
+        /** Standard positional markers */
+        @Query("update user set home_phone = ?, work_phone = ? where key = ?")
+        void updatePhones_positional(String homePhone, String workPhone, int key);
+
+        /** Standard named markers */
+        @Query("update user set home_phone = :home, work_phone = :work where key = :key")
+        void updatePhones_named(@Param("key") int key, @Param("home") String homePhone, @Param("work") String workPhone);
+
+        /** Named markers with no @Param. Should fallback to positional matching */
+        @Query("update user set home_phone = :home, work_phone = :work where key = :key")
+        void updatePhones_fallback(String homePhone, String workPhone, int key);
+
+        /** Named markers with repeated names */
+        @Query("update user set home_phone = :phone, work_phone = :phone where key = :key")
+        void updateBothPhones(@Param("key") int key, @Param("phone") String uniquePhone);
+
+        /** Named markers with repeated names, but no @Param. Should fallback to positional matching */
+        @Query("update user set home_phone = :phone, work_phone = :phone where key = :key")
+        void updatePhones_fallback2(String homePhone, String workPhone, int key);
+    }
+
+    @Accessor
+    public interface UserPhoneAccessor_NotEnoughParams {
+        @SuppressWarnings("unused")
+        @Query("update user set home_phone = :phone, work_phone = :phone where key = :key")
+        void updateBothPhones(@Param("key") int key);
+    }
+
+    public interface UserPhoneAccessor_TooManyParams {
```

Comment:
Missing `@Accessor` annotation here so should_fail_if_not_enough_parameters throws a different exception then expected (but still passes since its an instance of RuntimeException):

> java.lang.IllegalArgumentException: @Accessor annotation was not found on type com.datastax.driver.mapping.MapperAccessorParamsTest$UserPhoneAccessor_TooManyParams

Adding `@Accessor` gets it to fail with the right exception:

> Caused by: java.lang.IllegalArgumentException: Too many arguments for method updatePhones, found 4 but it should be at most the number of bind parameters in the @Query (3)